src/components/LoginDialog.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <q-card style="min-width: 25%">
+    <q-card-section>
+      <div class="text-h6">
+        Log in
+      </div>
+    </q-card-section>
+    <form @submit.prevent="onSubmit">
+      <q-card-section>
+        <q-input
+          v-model="username"
+          dense
+          outlined
+          placeholder="Username"
+          stack-label="Stack Label"
+          :rules="[val => val.length<=15 || 'Usernames should not be over 15 characters']"
+        />
+      </q-card-section>
+      <q-card-section>
+        <q-input
+          v-model="password"
+          type="password"
+          dense
+          outlined
+          placeholder="Password"
+          stack-label="Stack Label"
+          :rules="[val => val.length<=15 || 'Passwords should not be over 15 characters']"
+        />
+      </q-card-section>
+      <q-card-section
+        align="right"
+        class="text-primary row justify-around"
+      >
+        <q-btn
+          v-close-popup
+          flat
+          label="Log in"
+          type="submit"
+          :disable="!(username && password)"
+        />
+        <q-btn
+          v-close-popup
+          flat
+          label="Cancel"
+        />
+      </q-card-section>
+    </form>
+    <q-card-section>
+      <div>
+        New to IceMeme?
+        <router-link
+          v-close-popup
+          :to="{ name: 'register' }"
+        >
+          Sign up.
+        </router-link>
+      </div>
+    </q-card-section>
+  </q-card>
+</template>
+
+<script>
+export default {
+  name: 'LoginDialog',
+  data() {
+    return {
+      username: '',
+      password: '',
+    };
+  },
+  methods: {
+    onSubmit() {
+      if (this.username && this.password) {
```

Comment:
```suggestion
      if (this.readyToSubmit()) {
```

Reply:
I'm having trouble calling the 'computed' from the methods... 
readyToSubmit is returning the right value, but it's like I'm skipping the onSubmit method after that

Reply:
Couldn't solve this, sorry,
 so I'm only calling readyToSubmit from the template, not from the methods. 
My assumption is: it's already called once to even reach the onSubmit method, that's a good enough check.

Reply:
I believe I know how to fix this now - works fine as it is, but I will fix it later